core/src/main/scala/org/apache/spark/h2o/H2OConf.scala

```diff
@@ -107,7 +110,29 @@ class H2OConf(val sparkConf: SparkConf) extends Logging with InternalBackendConf
   }
 }
 
-object H2OConf {
+object H2OConf extends Logging {
+
+  private val deprecatedOptions = Map("spark.ext.h2o.external.cluster.num.h2o.nodes" -> ExternalBackendConf.PROP_EXTERNAL_CLUSTER_SIZE._1)
+
+  private def checkDeprecatedOptions(sparkConf: SparkConf): Unit = {
+
+    deprecatedOptions.foreach {
+      case (deprecated, current) =>
+        val deprecatedValue = sparkConf.getOption(deprecated)
+        if (deprecatedValue.isDefined) {
+          val currentValue = sparkConf.getOption(current)
+          if (currentValue.isDefined) {
+            logWarning(s"Both options '$deprecated' and '$current' are specified. " +
+              s"Using value '${currentValue.get}' of '$current' as the later one is deprecated.")
+          } else {
+            logWarning(s"Please use '$current' as '$deprecated' is deprecated. Passing the value '${deprecatedValue.get}' to '$current'.")
+          }
+          sparkConf.set(current, deprecatedValue.get)
```

Comment:
If both are specified, doesn't you override the current with a value from deprecated?

Reply:
Very good catch!